lock.go

```diff
@@ -13,35 +14,52 @@ import (
 const luaRefresh = `if redis.call("get", KEYS[1]) == ARGV[1] then return redis.call("pexpire", KEYS[1], ARGV[2]) else return 0 end`
 const luaRelease = `if redis.call("get", KEYS[1]) == ARGV[1] then return redis.call("del", KEYS[1]) else return 0 end`
 
+var ErrCanntGetLock = errors.New("can't get lock")
+
 // RedisClient is a minimal client interface
 type RedisClient interface {
 	SetNX(key string, value interface{}, expiration time.Duration) *redis.BoolCmd
 	Eval(script string, keys []string, args ...interface{}) *redis.Cmd
 }
 
+// Handler is method wrapper
+type Handler func() error
+
 // Lock allows distributed locking
 type Lock struct {
 	client RedisClient
 	key    string
-	opts   LockOptions
+	opts   Options
 
 	token string
 	mutex sync.Mutex
 }
 
+// RunWithLock run some code with Redis Lock
+func RunWithLock(client RedisClient, key string, handler Handler, opts *Options) error {
```

Comment:
could you pls change this to: `func Run(client RedisClient, key string, opts *Options, handler func() error) error`, i.e. rename func name and put the handler at the end?

Reply:
ok, no problems